extensions/amp-ad/0.1/amp-ad.js

```diff
@@ -394,7 +398,7 @@ class AmpAd extends AMP.BaseElement {
         this.toggleFallback(true);
       }
       // Remove the iframe only if it is not the master.
-      if (this.iframe_.name.indexOf('_master') == -1) {
+      if (this.iframe_ && this.iframe_.name.indexOf('_master') == -1) {
```

Comment:
I don't see why checking `this.iframe_` is not null here would change behaviors when user navigates back. 
Here's what I understand, the only reason `this.iframe_` is null here is that it's set in `unlayoutCallback()` before. And if that's the case we don't need to call `removeElement` and set`this.iframe_ = null` here. Also we are not doing anything when `this.iframe_ = null` here before.